File: 3336-water-bottles-ii/water-bottles-ii.cpp
class Solution {
public:
    int maxBottlesDrunk(int numBottles, int numExchange) {
        if(numBottles < numExchange){
            return numBottles;
        }
        int ans=numBottles;
        int remains=numBottles;

        while(remains >= numExchange){
            remains-=numExchange;
            numExchange++;
            remains++;
            ans++;
        }
        return ans;
    }
};

File: 3336-water-bottles-ii/water-bottles-ii.py
class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        drunk_bottles=numBottles
        while numBottles >= numExchange:
            numBottles-=numExchange
            numBottles+=1
            drunk_bottles+=1
            numExchange+=1
        return drunk_bottles
        